bootstrap/handlers/httpserver.go

```diff
@@ -139,10 +140,15 @@ func (b *HttpServer) BootstrapHandler(
 
 		b.isRunning = true
 		err := server.ListenAndServe()
-		if err != nil {
+		// "Server closed" error occurs when Shutdown above is called in the Done processing, so it can be ignored
+		if err != nil && err != http.ErrServerClosed {
+			// Other errors occur during bootstrapping, like port bind fails, are considered fatal
 			lc.Errorf("Web server failed: %v", err)
 			cancel := container.CancelFuncFrom(dic.Get)
-			cancel() // this will caused the service to stop
+			cancel() // this will clean up any long-running go functions that may have started
+			// Give time for clean up to occur before exiting.
+			time.Sleep(1 * time.Millisecond)
```

Comment:
why 1 millis sleep? Is this a trick/hack? Does `cancel()` return before completed?

Reply:
Yes, a hack because `cancel()` write to the channel and returns before receivers have completed.
But this made me think and realized we have the wait group to use instead of sleep.